primitives/core/src/assertion/mod.rs

```diff
@@ -164,7 +167,11 @@ impl Assertion {
 			Self::VIP3MembershipCard(..) |
 			Self::WeirdoGhostGangHolder => vec![Web3Network::Ethereum],
 			// total tx over `networks`
-			Self::A8(network) => network.to_vec(),
+			Self::A8(networks) => networks
```

Comment:
How do you think just add a filter to filter out types, which is NOT `Litentry, Litmus, Polkadot, Kusama, Khala, or Ethereum` ?
Then we don't need to create a new type `A8SupportedNetwork`.

Reply:
Does the client pass in the wrong networks? Shouldn't this be fixed at client side first?

But sure we could a second check logic here

And yes we can simply filter out unwanted types, or `retain` only desired ones

Reply:
@Kailai-Wang this is the leftover of refactoring of shifting network type from `link_identity` to `request_vc`. From IDHub point of view, they always wish the all supported networks. So they don't do any filtering at client side. We did the magic within `get_eligible_identities()`.

Reply:
refactored to only filter instead creating new types.

Reply:
> this is the leftover of refactoring of shifting network type from link_identity to request_vc. From IDHub point of view, they always wish the all supported networks. So they don't do any filtering at client side. We did the magic within get_eligible_identities().

I doubt that's related.

The main motivation here was to have a struct to represent exactly the accepted values for A8. Clients must specify one of them to get the right assertion.